src/main/java/me/coley/recaf/util/Classpath.java

```diff
@@ -138,38 +146,112 @@ public void scan(ClassLoader classLoader) {
+						verifyScan();
 					} catch (ReflectiveOperationException | SecurityException e) {
 						throw new ExceptionInInitializerError(e);
 					}
 				} else {
-					// The internal implementation, including the class loading mechanism,
-					// was completely redesigned in Java 9. And after some hours of research,
-					// I believe that it was theoretically impossible to acquire the bootstrap classpath anymore -
-					// it seems to be completely native, and doesn't show up anywhere in the code.
-					Logging.warn("Recaf cannot acquire the classpath for bootstrap classes when running in Java 9 or above. " +
-							"This will affect auto-complete suggestions in Assembler!");
+					try {
+
+						// Before we will do that, break into Jigsaw module system to grant full access
+						Class<?> moduleClass = Class.forName("java.lang.Module");
+						Class<?> layer = Class.forName("java.lang.ModuleLayer");
+						Field lookupField = MethodHandles.Lookup.class.getDeclaredField("IMPL_LOOKUP");
+						lookupField.setAccessible(true);
+						MethodHandles.Lookup lookup = (MethodHandles.Lookup) lookupField.get(null);
+						MethodHandle export = lookup
+								.findVirtual(moduleClass, "implAddOpens", MethodType.methodType(void.class, String.class));
+						MethodHandle getPackages = lookup
+								.findVirtual(moduleClass, "getPackages", MethodType.methodType(Set.class));
+						MethodHandle getModule = lookup
+								.findVirtual(Class.class, "getModule", MethodType.methodType(moduleClass));
+						MethodHandle getLayer = lookup
+								.findVirtual(moduleClass, "getLayer", MethodType.methodType(layer));
+						MethodHandle layerModules = lookup
+								.findVirtual(layer, "modules", MethodType.methodType(Set.class));
+						MethodHandle unnamedModule = lookup
+								.findVirtual(ClassLoader.class, "getUnnamedModule", MethodType.methodType(moduleClass));
+						Set modules = new HashSet();
+
+						Object ourModule = getModule.invoke(Classpath.class);
+						Object ourLayer = getLayer.invoke(ourModule);
+						if (ourLayer != null) {
+							modules.addAll((Set) layerModules.invoke(ourLayer));
+						}
+						modules.addAll(
+								(Set) layerModules
+										.invoke(lookup.findStatic(layer, "boot", MethodType.methodType(layer))
+												.invoke()));
+						for (ClassLoader c = Classpath.class.getClassLoader(); c != null; c = c.getParent()) {
+							modules.add(unnamedModule.invoke(c));
+						}
+
+						for (Object impl : modules) {
+							for (String name : (Set<String>) getPackages.invoke(impl)) {
+								export.invoke(impl, name);
+							}
+						}
+
+						Class<?> reflectionClass = Class.forName("jdk.internal.reflect.Reflection");
+						MethodHandle getMapHandle = lookup.findStaticGetter(reflectionClass, "fieldFilterMap", Map.class);
+						// Map is immutable, thanks Oracle
+						Map<Class<?>, Set<String>> fieldFilterMap = new HashMap<>((Map<Class<?>, Set<String>>) getMapHandle.invokeExact());
```

Comment:
```suggestion
						Map<Class<?>, ?> fieldFilterMap = new HashMap<>((Map<Class<?>, ?>) getMapHandle.invokeExact());
```

They changed some code in `Reflection` in JDK 12: [before](https://github.com/openjdk/jdk11u/blob/71f32ecc0fa1d0c552c40b1222412a70678dea19/src/java.base/share/classes/jdk/internal/reflect/Reflection.java#L46-L58) and  [after](https://github.com/openjdk/jdk12u/blob/fca351466233b0ac5867f3cd690897df3d707726/src/java.base/share/classes/jdk/internal/reflect/Reflection.java#L45-L63), so 

1. The generic signature of `fieldFilterMap` should be `Map<Class<?>, ?>` for compatibility between `Map<Class<?>, String[]>` and `Map<Class<?>, Set<String>>` (we are only using the key anyway).
2. What's more, it's altogether unnecessary to do this before Java 12 - older versions didn't filter out `Field` access.